src/backend/distributed/planner/distributed_planner.c

```diff
@@ -105,13 +106,15 @@ distributed_planner(Query *parse, int cursorOptions, ParamListInfo boundParams)
 
 	if (needsDistributedPlanning)
 	{
+		fastPathRouterQuery = FastPathRouterQuery(parse);
+
 		/*
 		 * Inserting into a local table needs to go through the regular postgres
 		 * planner/executor, but the SELECT needs to go through Citus. We currently
 		 * don't have a way of doing both things and therefore error out, but do
 		 * have a handy tip for users.
 		 */
-		if (InsertSelectIntoLocalTable(parse))
+		if (!fastPathRouterQuery && InsertSelectIntoLocalTable(parse))
```

Comment:
wondering whether there is a nicer place for the InsertSelectIntoLocalTable check